tests/test_bin.py

```diff
@@ -623,3 +631,21 @@ def test_hystartup():
     assert "[1,~2]" in output
 
     del os.environ["HYSTARTUP"]
+
+
+def test_output_buffering(tmp_path):
+    tf = tmp_path / "file.txt"
+
+    pf = tmp_path / "program.hy"
+    pf.write_text(f'''
+        (print "line 1")
+        (import  sys  pathlib [Path])
+        (print :file sys.stderr (.strip (.read-text (Path #[=[{tf}]=]))))
```

Comment:
placing the text after the keyword argument shouldn't be legal. it should crash with `positional argument follows keyword argument` as it does in python. Unrelated to this PR, but we should open an issue for this.

Reply:
I've always seen this as a feature, and went out of my way to make it work for method calls (#1167). A lot of times we can't effectively get around Python limitations, but it's nice when we can.